core/src/main/java/com/oath/oak/NativeAllocator/OakNativeMemoryAllocator.java

```diff
@@ -18,11 +18,18 @@
 
 public class OakNativeMemoryAllocator implements OakMemoryAllocator {
 
+    // When allocating n bytes and there are buffers in the free list, only free buffers of size <= n * RECLAIM_FACTOR will be recycled
+    // This parameter may be tuned for performance vs off-heap memory utilization
+    private static final int RECLAIM_FACTOR = 2;
+
     // blocks allocated solely to this Allocator
     private final ConcurrentLinkedQueue<Block> blocks = new ConcurrentLinkedQueue<>();
-    // free list of ByteBuffers which can be reused
-    private final ConcurrentSkipListSet<Pair<Integer,ByteBuffer>> freeList =
-        new ConcurrentSkipListSet<>(Comparator.comparing(Pair::getKey));
+
+    // free list of ByteBuffers which can be reused - sorted by buffer size, then by unique hash
+    private Comparator<Pair<Integer, ByteBuffer>> comparator =
+            Comparator.<Pair<Integer, ByteBuffer>, Integer>comparing(p -> p.getValue().capacity())
+            .thenComparing(Pair::getKey);
+    private final ConcurrentSkipListSet<Pair<Integer,ByteBuffer>> freeList = new ConcurrentSkipListSet<>(comparator);
```

Comment:
Why do we need the Pair? Why not ask for identity while comparing? Is it slower?